src/Broadway/EventStore/MongoDBEventStore.php

```diff
@@ -0,0 +1,367 @@
+     * @param SerializerInterface    $payloadSerializer
+     * @param SerializerInterface    $metadataSerializer
+     * @param UuidGeneratorInterface $uuidGenerator
+     * @param string                 $eventCollectionName
+     * @param string                 $transactCollectionName
+     */
+    public function __construct(
+        Database $database,
+        SerializerInterface $payloadSerializer,
+        SerializerInterface $metadataSerializer,
+        UuidGeneratorInterface $uuidGenerator,
+        $eventCollectionName,
+        $transactCollectionName
+    ) {
+        $this->payloadSerializer    = $payloadSerializer;
+        $this->metadataSerializer   = $metadataSerializer;
+        $this->uuidGenerator        = $uuidGenerator;
+
+        $this->eventCollection      = $database->selectCollection($eventCollectionName);
+        $this->eventCollection->setReadPreference(MongoClient::RP_PRIMARY);
+
+        $this->transactCollection   = $database->selectCollection($transactCollectionName);
+        $this->transactCollection->setReadPreference(MongoClient::RP_PRIMARY);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function load($id)
+    {
+        $cursor = $this->eventCollection
+            ->find(array(
+                self::FIELD_STREAM_ID => (string) $id,
+                self::FIELD_COMMIT_ID => null,
+            ), array(
+                '_id' => 0,
+            ))
+            ->sort(array(self::FIELD_PLAYHEAD => 1))
+        ;
+
+        $events = array();
+
+        foreach ($cursor as $row) {
+            $events[] = $this->deserializeEvent($row);
+        }
+
+        if (!$events) {
+            throw new EventStreamNotFoundException(sprintf('EventStream not found for aggregate with id %s', $id));
+        }
+
+        return new DomainEventStream($events);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function append($id, DomainEventStreamInterface $eventStream)
+    {
+        $aggregateId = (string) $id;
+
+        $events = $this->extractEvents($eventStream, $aggregateId);
+
+        if (!$events) {
+            return;
+        }
+
+        if (count($events) == 1) {
+            try {
+                $this->saveEvent(reset($events));
+            } catch (MongoException $exception) {
+                throw MongoDBEventStoreException::create($exception);
+            } catch (Exception $exception) {
+                throw new MongoDBEventStoreException($exception->getMessage(), 0, $exception);
+            }
+
+            return;
+        }
+
+        try {
+            $this->saveBulkEvents($events);
+        } catch (MongoException $exception) {
+            throw MongoDBEventStoreException::create($exception);
+        } catch (Exception $exception) {
+            throw new MongoDBEventStoreException($exception->getMessage(), 0, $exception);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureCollection()
```

Comment:
It's done in the same way as DBAL implementation, it requires indices to be set before processing.